### PR TITLE
Candidate Test Plans page row consolidation

### PR DESCRIPTION
**Pull request by howard-e**
This PR aims to merge multiple Test Plan reports into a single row where applicable (the same logic is being applied to the Test Reports page)

**NB. Currently based on #499**

### CODE REVIEW

File: client/components/TestQueueRow/index.jsx [hunk 1]
```diff
@@ -378,14 +385,107 @@ const TestQueueRow = ({
...
+                    const { data } = await client.query({
+                        query: TEST_PLAN_REPORT_CANDIDATE_RECOMMENDED_QUERY,
+                        fetchPolicy: 'network-only'
+                    });
+
+                    // TODO: Check to see if the proposed test plan report to be promoted to CANDIDATE
+                    //  has a test plan version date less than whatever is in CANDIDATE or RECOMMENDED
+                    //  tests already, if any, and note it may never be displayed to the admin
+
+                    // --- SECTION START: OutdatedCandidatePhaseComparison
```
Comment (evmiguel) [inline client/components/TestQueueRow/index.jsx:400]: Thank you for this documentation!

**Review by alflennik — APPROVED** [review]
I can confirm that this works and that the behavior seems pretty logical from what I understand about the workflow, and also I want to note that the new UI that I hadn't seen before is looking pretty nice and seems easy to use. 

Looking at all the imperative-style code, one opportunity here might be to eventually write an integration test which captures the complete end-to-end flow of a test plan through its entire life cycle, because to me it's hard to really feel like I tested it thoroughly enough and caught all the potential edge cases, but I don't think that's a blocker for this PR.

Nice work!

**Review by evmiguel — APPROVED** [review]
Thanks for taking care of the Candidate Test Run page and great documentation for the modal fix! This was a big body of work done over a short period of time, so kudos to you!
